ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/IteratingLinearOptimizer.java

```diff
@@ -243,6 +233,35 @@ private static Pair<IteratingLinearOptimizationResultImpl, Boolean> updateBestRe
         return Pair.of(bestResult, !raRangeShrinking);
     }
 
+    private static RangeActionActivationResult roundResult(RangeActionActivationResult linearProblemResult, IteratingLinearOptimizationResultImpl previousResult, IteratingLinearOptimizerInput input, IteratingLinearOptimizerParameters parameters) {
+        RangeActionActivationResultImpl roundedResult = new RangeActionActivationResultImpl(input.getPrePerimeterSetpoints());
+        if (parameters.getRangeActionParameters().getPstModel().equals(RangeActionsOptimizationParameters.PstModel.CONTINUOUS)) {
+            BestTapFinder.round(
+                linearProblemResult,
+                input.getNetwork(),
+                input.getOptimizationPerimeter(),
+                previousResult,
+                parameters.getObjectiveFunctionUnit(), roundedResult
+            );
+        } else {
+            input.getOptimizationPerimeter().getRangeActionOptimizationStates().forEach(state -> linearProblemResult.getActivatedRangeActions(state).forEach(rangeAction -> {
+                if (rangeAction instanceof PstRangeAction pstRangeAction) {
+                    roundedResult.activate(pstRangeAction, state, pstRangeAction.convertTapToAngle(linearProblemResult.getOptimizedTap(pstRangeAction, state)));
+                }
+            }));
+        }
```

Comment:
put all this in a new "roundPsts"  method, so that "roundOtherRas" name becomes understandable